### PR TITLE
refactor(NetworkManagerGUI): prevent potential NRE spam if reference is lost and tidy up

### PR DESCRIPTION
**Pull request by SoftwareGuy**
While chasing a ghost error due to Unity adding two NetworkManagerGUIs to one object and I forgot to check that, I discovered that the component and spam NREs in its OnGUI loop.

So, what this PR does is quite a fair bit actually...

1. Short-circuits if there's no NetworkManager component specified and _Auto Configure_ is turned off. This prevents crap loads of NRE spam due to OnGUI trying to access stuff on a null object.
2. It introduces a new "Auto Configure" option which means people can slap on the component, tick a box and it'll automatically reference NetworkManager should it exist on the same GameObject. If it doesn't, it will throw an `ArgumentNullException` and tell them to fix it.  However, I have implemented it so if you do specify a NetworkManager reference and leave it ticked to automatically configure itself that it will not override your manual reference.
3. Adds tooltips and explanations as well as adds headers to the configuration options.
4. Blows off a ton of dust and gives the code a restructure with regions, etc.

As an added bonus, it also addresses Visual Studio complaining about IDE0048: "Add parentheses for clarity" by... you name it, adding parentheses around some Rect math.

Mirage. Makin' NetworkManageGUI great again.

### CODE REVIEW

**Comment by SoftwareGuy** [comment]
Example of what it looks like now: 

image

File: Assets/Mirage/Components/NetworkManagerGUI.cs [hunk 1]
```diff
@@ -1,18 +1,70 @@
...
+            if (AutoConfigureNetworkManager)
+            {
+                if (NetworkManager == null)
+                {
+                    NetworkManager = GetComponent<NetworkManager>();
+
+                    // Is this STILL null? Then we throw in the towel and go home.
+                    if (NetworkManager == null)
+                    {
+                        throw new ArgumentNullException("NetworkManager", $"You requested automatic configuration for the NetworkManagerGUI component on '{gameObject.name}'" +
```
Comment (James-Frowen) [inline Assets/Mirage/Components/NetworkManagerGUI.cs:43]: this should be `InvalidOperationException` instead of Argument

File: Assets/Mirage/Components/NetworkManagerGUI.cs [hunk 2]
```diff
@@ -1,18 +1,70 @@
...
+
+        [Tooltip("Select where you want the NetworkManagerGUI elements to be located on the screen.")]
         public TextAnchor GUIAnchor = TextAnchor.UpperLeft;
 
+        private void Awake()
+        {
+            // Coburn, 2023-01-29: 
+            // If automatic configuration of NetworkManager is enabled, then attempt to grab the
+            // NetworkManager that this script is attached to.
+            if (AutoConfigureNetworkManager)
```
Comment (James-Frowen) [inline Assets/Mirage/Components/NetworkManagerGUI.cs:34]: if it hard to follow this chain of if/else,, would be better to do early returns instead

File: Assets/Mirage/Components/NetworkManagerGUI.cs [hunk 3]
```diff
@@ -1,18 +1,70 @@
...
+                    {
+                        throw new ArgumentNullException("NetworkManager", $"You requested automatic configuration for the NetworkManagerGUI component on '{gameObject.name}'" +
+                            $" but one could not be found. Either disable automatic configuration or ensure this script is on a GameObject with a Mirage NetworkManager.");
+                    }
+                }
+                else
+                {
+                    if (logger.logEnabled)
+                    {
+                        logger.LogWarning($"You have enabled Auto Configure for this NetworkManagerGUI component on '{gameObject.name}' but the NetworkManager field is already set. " +
```
Comment (James-Frowen) [inline Assets/Mirage/Components/NetworkManagerGUI.cs:51]: I dont think we need this warning.
